src/query/query_manager.c

```diff
@@ -2186,26 +2134,18 @@ qmgr_clear_trans_wakeup (THREAD_ENTRY * thread_p, int tran_index, bool is_tran_d
       XASL_ID_SET_NULL (&query_p->xasl_id);
 
       /* end use of the list file of the cached result */
-      if (!query_p->is_holdable || is_abort)
+      if (query_p->xasl_ent != NULL && query_p->list_ent != NULL)
 	{
-	  if (query_p->xasl_ent != NULL && query_p->list_ent != NULL)
-	    {
-	      (void) qfile_end_use_of_list_cache_entry (thread_p, query_p->list_ent, false);
-	    }
-	  query_p->query_status = QUERY_CLOSED;
+	  (void) qfile_end_use_of_list_cache_entry (thread_p, query_p->list_ent, false);
+	  query_p->xasl_ent = NULL;
```

Comment:
where query_p->xasl_ent is deallocated?

Reply:
let me check of it, it looks something wrong.